### PR TITLE
Different description in overwrite dialog when versioning is enabled

### LINKED ISSUE #3047: Overwrite dialog wording
### Steps

1. Upload a file "test.txt"
1. Upload it again

### Expected

- if versions app enabled, the text should say "Create new version"
- otherwise it should say "Overwrite"

cc @pmaier1

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description
If versioning is enabled, display "Do you want to create new version?" message.

## Related Issue

- Fixes #3047

## How Has This Been Tested?

- test environment: Drone

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by LukasHirt** [comment]
The wording might be improved. E.g. instead of "want" "wish" but that would require touching also the message for overwrite so I think it should be addressed separately.

**Comment by LukasHirt** [comment]
> But file "lorem (2).txt" should not be listed on the webUI - expected "visible" but got: "not visible"

Something is strange here...

File: apps/files/src/mixins.js [hunk 1]
```diff
@@ -152,7 +152,13 @@ export default {
           } else {
             const translated = this.$gettext('File %{file} already exists.')
             this.setOverwriteDialogTitle(this.$gettextInterpolate(translated, { file: item.name }, true))
-            this.setOverwriteDialogMessage(this.$gettext('Do you want to overwrite it?'))
+
+            if (this.capabilities.files.versioning) {
+              this.setOverwriteDialogMessage(this.$gettext('Do you want to create new version?'))
```
Comment (butonic) [inline apps/files/src/mixins.js]: Do you want to create _a_ new version

File: apps/files/src/mixins.js [hunk 2]
```diff
@@ -177,7 +183,13 @@ export default {
 
         const translated = this.$gettext('File %{file} already exists.')
         this.setOverwriteDialogTitle(this.$gettextInterpolate(translated, { file: files[i].name }, true))
-        this.setOverwriteDialogMessage(this.$gettext('Do you want to overwrite it?'))
+
+        if (this.capabilities.files.versioning) {
+          this.setOverwriteDialogMessage(this.$gettext('Do you want to create new version?'))
```
Comment (butonic) [inline apps/files/src/mixins.js]: Do you want to create **a** new version

File: changelog/unreleased/3047 [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Enhancement: Different message in overwrite dialog when versioning is enabled
+
+We've added a new message in the overwrite dialog when versioning is enabled.
+This message is intended to make it clear that the resource won't be overwritten but new version of it will be created.
```
Comment (butonic) [inline changelog/unreleased/3047]: **a** new version

**Comment by LukasHirt** [comment]
Agh, now I understand the error. Since I put there the check for versions which is not last, on smaller screens it covers the whole files list 🤦‍♂

File: changelog/unreleased/3047 (hunk 3 above)
Comment (LukasHirt) [reply changelog/unreleased/3047]: Thanks for spotting, fixed 👍

File: apps/files/src/mixins.js (hunk 2 above)
Comment (LukasHirt) [reply apps/files/src/mixins.js]: Thanks for spotting, fixed 👍

File: apps/files/src/mixins.js (hunk 1 above)
Comment (LukasHirt) [reply apps/files/src/mixins.js]: Thanks for spotting, fixed 👍

**Pull request merged**